### ISSUE TITLE
caseInsensitive error: expected a Terminal (e.g. "abc")

### ISSUE THREAD

**Issue by htor** [harc/ohm#246]
ohm-js version: 0.14.0

i got an error after moving my working grammar from the online ohm-editor to my node program. 
when running the following in node:

```js
// test.js
const ohm = require('ohm-js')

ohm.grammar(`
Test {
  Match = i<"foo">
  i<x> = caseInsensitive<x>
}
`)
```

 it fails with the error, complaining about the rule `i<x>`, apparently:

```
~/Programming/sceditor $ node test.js
/Users/n642275/Programming/sceditor/node_modules/ohm-js/src/errors.js:233
      throw errors[0];
      ^

Error: Incorrect argument type: expected a Terminal (e.g. "abc")
    at createError (/Users/n642275/Programming/sceditor/node_modules/ohm-js/src/errors.js:20:9)
    at Object.incorrectArgumentType (/Users/n642275/Programming/sceditor/node_modules/ohm-js/src/errors.js:143:10)
    at Apply.pexprs.Apply._assertAllApplicationsAreValid (/Users/n642275/Programming/sceditor/node_modules/ohm-js/src/pexprs-assertAllApplicationsAreValid.js:98:20)
    at Apply.pexprs.PExpr.assertAllApplicationsAreValid (/Users/n642275/Programming/sceditor/node_modules/ohm-js/src/pexprs-assertAllApplicationsAreValid.js:24:8)
    at /Users/n642275/Programming/sceditor/node_modules/ohm-js/src/GrammarDecl.js:115:12
    at Array.forEach (<anonymous>)
    at GrammarDecl.build (/Users/n642275/Programming/sceditor/node_modules/ohm-js/src/GrammarDecl.js:107:30)
    at Semantics.Wrapper.Grammar (/Users/n642275/Programming/sceditor/node_modules/ohm-js/src/main.js:85:20)
    at Operation.doAction (/Users/n642275/Programming/sceditor/node_modules/ohm-js/src/Semantics.js:659:16)
    at Operation.execute (/Users/n642275/Programming/sceditor/node_modules/ohm-js/src/Semantics.js:627:18)
```

replacing the application of `i<"foo">` with righthand `caseInsensitive<"foo">` is a workaround the issue.

it works in the online editor, which doesn't complain about this. might be that the hosted ohm-js script version is newer/older than the npm package?

**Comment by htor** [comment]
@alexwarth @pdubroy 
i see the npm version is 0.14 and the latest version on github is 0.15. 
would you mind `npm publish` the latest version?

**Comment by ashnur** [comment]
dead project

**Comment by htor** [comment]
why is it though? last commit is from february this year, so i didn't assume it was dead? sorry to see a waste of good energy.. in these great, fast-changing, fast-forward times

**Comment by ashnur** [comment]
I wish I knew. But it's quite obvious when the npm is  not updated and issues remain unanswered for months that something is fundamentally broken here. Maybe work goes on elsewhere?

**Comment by ashnur** [comment]
OTOH. you can install the github version from npm with just `npm i harc/ohm` I think.

**Comment by alexwarth** [comment]
Hi htor,

The issue is that the `caseInsensitive` rule is not actually a rule. In fact, if you look at the code you'll see that we called it "case insensitive terminal". It exists to give the author of a grammar a way to express that a terminal is not case sensitive (which is what you get by default in Ohm). At the time, @pdubroy and I spent considered just adding a different syntax for case-insensitive terminals, but in the end we decided on this pseudo-rule because it is more explicit. The down side is that the fact that it's not a full-fledged rule is also confusing, as you've discovered. Sorry about that!

And to address ashnur's comment: yes, it's been some time since we last made any changes to the code, or published a new version of the npm. But this project is certainly not dead! I for one am still using Ohm in a bunch of different projects.

Best,
Alex
